packages/cactus-cmd-api-server/src/test/typescript/unit/get-open-api-spec-v1-oauth2-scopes.test.ts

```diff
@@ -0,0 +1,239 @@
+    );
+  });
+
+  it("HTTP - allows health check execution with a valid JWT Token", async () => {
+    const jwtPayload = { scope: "read:health" };
+    const validJwt = await new SignJWT(jwtPayload)
+      .setProtectedHeader({ alg: "RS256" })
+      .setIssuer(expressJwtOptions.issuer)
+      .setAudience(expressJwtOptions.audience)
+      .sign(jwtKeyPair.privateKey);
+
+    const validBearerToken = `Bearer ${validJwt}`;
+    expect(validBearerToken).toBeTruthy();
+
+    const resPromise = apiClient.getHealthCheckV1({
+      headers: { Authorization: validBearerToken },
+    });
+
+    await expect(resPromise).resolves.toHaveProperty("data");
+    const res = await resPromise;
+    expect(res.status).toEqual(200);
+    expect(res.data).toBeTruthy();
+  });
+
+  it("HTTP - rejects health check execution with an invalid JWT", async () => {
+    const { privateKey: otherPrivateKey } = await generateKeyPair("RS256");
+    const invalidJwt = await new SignJWT({ scope: "invalid:scope" })
+      .setProtectedHeader({ alg: "RS256" })
+      .setIssuer("invalid-issuer")
+      .setAudience("invalid-audience")
+      .sign(otherPrivateKey);
+
+    const invalidBearerToken = `Bearer ${invalidJwt}`;
+    expect(invalidBearerToken).toBeTruthy();
+
+    const resPromise = apiClient.getHealthCheckV1({
+      headers: { Authorization: invalidBearerToken },
+    });
+    await expect(resPromise).rejects.toThrow(
+      "Request failed with status code 401",
+    );
+  });
+
+  it("HTTP - allows Prometheus metrics execution with a valid JWT Token", async () => {
+    const jwtPayload = { scope: "read:metrics" };
+    const validJwt = await new SignJWT(jwtPayload)
+      .setProtectedHeader({ alg: "RS256" })
+      .setIssuer(expressJwtOptions.issuer)
+      .setAudience(expressJwtOptions.audience)
+      .sign(jwtKeyPair.privateKey);
+
+    const validBearerToken = `Bearer ${validJwt}`;
+    expect(validBearerToken).toBeTruthy();
+
+    const resPromise = apiClient.getPrometheusMetricsV1({
+      headers: { Authorization: validBearerToken },
+    });
+
+    await expect(resPromise).resolves.toHaveProperty("data");
+    const res = await resPromise;
+    expect(res.status).toEqual(200);
+    expect(res.data).toBeTruthy();
+  });
+
+  it("HTTP - rejects Prometheus metrics execution with an invalid JWT", async () => {
+    const { privateKey: otherPrivateKey } = await generateKeyPair("RS256");
+    const invalidJwt = await new SignJWT({ scope: "invalid:scope" })
+      .setProtectedHeader({ alg: "RS256" })
+      .setIssuer("invalid-issuer")
+      .setAudience("invalid-audience")
+      .sign(otherPrivateKey);
+
+    const invalidBearerToken = `Bearer ${invalidJwt}`;
+    expect(invalidBearerToken).toBeTruthy();
+
+    const resPromise = apiClient.getPrometheusMetricsV1({
+      headers: { Authorization: invalidBearerToken },
+    });
+    await expect(resPromise).rejects.toThrow(
+      "Request failed with status code 401",
+    );
```

Comment:
@aldousalvarez Almost there! Please make sure to explicitly assert that the `status` property of the is 401 or 403 depending on which one is applicable (403 if the token is valid BUT missing the correct scope, 401 otherwise)

Reply:
@aldousalvarez Thank you for the updates!